### PR TITLE
Fix ice_getConnectionAsync

### LINKED ISSUE #1729: getConnectionAsync throws ConnectionClosed synchronously
See `TODO` in python Ice/ami test for `getConnectionAsync`, this exception was previously throw asynchronously when python extension used the old C++98 mapping.

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR fixes ice_getConnectionAsync to no throw synchronously NoEndpointException and other local exceptions.

Fixes #1729.

### CODE REVIEW

File: cpp/src/Ice/OutgoingAsync.cpp [hunk 1]
```diff
@@ -1064,11 +1062,7 @@ OutgoingAsync::invoke(string_view operation)
         return;
     }
 
-    //
-    // NOTE: invokeImpl doesn't throw so this can be called from the
```
Comment (bernardnormier) [inline cpp/src/Ice/OutgoingAsync.cpp:1068]: I don't know why we had this bogus comment everywhere.

File: csharp/src/Ice/Proxy.cs [hunk 2]
```diff
@@ -1583,14 +1583,7 @@ public override void handleInvokeResponse(bool ok, OutgoingAsyncBase og)
     private void iceI_ice_getConnection(OutgoingAsyncCompletionCallback completed, bool synchronous)
     {
         var outgoing = new ProxyGetConnection(this, completed);
-        try
-        {
-            outgoing.invoke(_ice_getConnection_name, synchronous);
-        }
-        catch (Ice.Exception ex)
-        {
-            outgoing.abort(ex);
```
Comment (bernardnormier) [inline csharp/src/Ice/Proxy.cs:1592]: The C# code was overall correct: we were catching and handling the exception with abort.

I just move this catch/handle to the invoke method implementation, for consistency with other invoke implementations and other language mappings.

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/ProxyGetConnection.java [hunk 3]
```diff
@@ -40,6 +40,10 @@ public int invokeCollocated(CollocatedRequestHandler handler) {
     }
 
     public void invoke() {
-        invokeImpl(true); // userThread = true
+        try {
```
Comment (bernardnormier) [inline java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/ProxyGetConnection.java:43]: This try/catch was missing in Java, just like in C++.

**Review by pepone — APPROVED** [review]
Looks good.

**Review by externl — APPROVED** [review]

**Pull request merged**